baseapp/msg_service_router.go

```diff
@@ -19,8 +19,13 @@ import (
 // MessageRouter ADR 031 request type routing
 // https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-031-msg-service.md
 type MessageRouter interface {
+	gogogrpc.Server
```

Comment:
BaseApp uses the methods on the concrete type, so yes they need to be there.

Reply:
tbh i think this is a bandaid to quickly close the issue. Im not sure it worth it as its just adding everything to an interface instead of defining a interface boundary between baseapp and the msg router. Id rather leave it as is then expand an interface just because

Reply:
for transparency i looked and made the changes locally myself then thought this isnt making the situation better, actually i thought it made the situation worse. This felt like a code smell. With some of the refactors coming up i decided not to push on it. I just commented on the issue.

Reply:
I do not like the setters in the interface, but when I looked at the issue some time ago, refactoring the interface registry setter of baseapp was leading to more breakage (circuit didn't exist then).
I can go that way, where it will clean it up a bit, but that means at message server instantiation you'll need to provide everything.